api/controllers/Irc_EntryController.js

```diff
@@ -55,5 +108,53 @@ module.exports = {
       .catch((error) => {
         res.serverError(error.message);
       });
+  },
+
+  process_event: function (request_body) {
+    switch (request_body.operation) {
+    case Event.OPERATION_INTER_SITE_TRANSFER:
+      return this.handleInterSiteTransfer(request_body);
+    case Event.OPERATION_UPDATE:
+      return processEventDetainee(request_body);
+    case Event.OPERATION_CHECK_IN:
+    case Event.OPERATION_CHECK_OUT:
+    case Event.OPERATION_REINSTATEMENT:
+      return processEventDetainee(request_body)
+        .then((detainee) => Event.create(generateStandardEvent(detainee, request_body)));
+    default:
+      throw new ValidationError('Unknown operation');
```

Comment:
You won't get here with an 'invalid operation' so it's not a validation error, it's just one that we don't handle yet.
ultimately you should return a `501 Not Implemented`